### PR TITLE
user and friend endpoints

### PR DESCRIPTION
**Pull request by M7s7**
This PR adds user and friend endpoints and the related test suite.

### CODE REVIEW

**Review by Rayahhhmed — APPROVED** [review]

**Comment by Rayahhhmed** [comment]
amazing job king 
wtf bro 11k lines of code we are cooked

File: server/prisma/schema.prisma [hunk 1]
```diff
@@ -74,7 +76,7 @@ model Timetable {
   selectedClasses Class[]
   createdEvents   Event[]
   user            User[]
-  userID          String
+  // userID          String
```
Comment (Shzmj) [inline server/prisma/schema.prisma]: oop can we remove these comments

File: server/prisma/schema.prisma [hunk 2]
```diff
@@ -109,9 +111,16 @@ enum GroupVisibility {
...
+  T2
+  T3
+}
+
+// enum ClassType { 
+//   LECTURE
+//   TUTORIAL
+//   LABORATORY
+//   OTHER
+// }
```
Comment (Shzmj) [inline server/prisma/schema.prisma]: oop can we remove these comments if not needed anymore

File: server/src/user/dto/timetable.dto.ts [hunk 3]
```diff
@@ -3,23 +3,58 @@ import { Type } from 'class-transformer';
...
+    enrolments: number;
+    capacity: number;
+  };
+  termDates: {
+    start: string;
+    end: string;
+  };
+  needsConsent: boolean;
+  mode: string;
+  times: any[]; // Change later but im lazy
```
Comment (Shzmj) [inline server/src/user/dto/timetable.dto.ts]: can we define proper type for this

File: server/test/db.spec.ts [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+      expect(res.status).not.toEqual(200);
+      expect(res.status).not.toEqual(201);
+    });
+
+    // This one will return an empty array instead of an error. This might actually be easier to work with
+    // test(`Timetable (user doesn't exist)`, async () => {
+    //   res = await request(app.getHttpServer()).get('/user/timetable/1');
+    //   expect(res).toEqual({});
+    //   expect(res.status).not.toEqual(200);
+    // });expect(res.status).not.toEqual(201);
```
Comment (Shzmj) [inline server/test/db.spec.ts]: can you check whats wrong with this test case

File: server/test/db.spec.ts [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+    //     .post('/friend')
+    //     .send({ senderId: user1.userID, sendeeId: user2.userID });
+    //   expect(res.status).toEqual(201);
+
+    //   res = await request(app.getHttpServer())
+    //     .post('/friend/request')
+    //     .send({ senderId: user1.userID, sendeeId: user2.userID });
+    //   expect(res.status).not.toEqual(200);
+    //   expect(res.status).not.toEqual(201);
+    // });
```
Comment (Shzmj) [inline server/test/db.spec.ts:495]: check whats wrong with the edge cases

**Review by Shzmj — CHANGES_REQUESTED** [review]

**Review by Shzmj — APPROVED** [review]

**Comment by Shzmj** [comment]
ready to merge 🙈
